engine/pom.xml

```diff
@@ -229,10 +229,10 @@
                             javax.persistence.*;version="[2.0.0,3.0.0)";resolution:=optional,
                             javax.validation.*;version="[1.0.0,2.0.0)",
                             javax.xml.*;version="0",
-                            javax.el.*;version="[2.0.0,4.0.0)",
+                            javax.el.*;version="[2.0.0,4.0.0)";resolution:=optional,
                             org.xml.sax.*;version="0",
                             org.jboss.logging.*;version="[3.1.0,4.0.0)",
-                            com.fasterxml.classmate.*;version="0.8.0",
+                            com.fasterxml.classmate.*;version="[${classmate.version},2.0.0)",
```

Comment:
> Isn't the whole point of OSGi that - being modular - different modules can depend on different versions?

Sure, you can have different versions of the same module if needed. But of course its still nicer to only work with one version of possible - less classes loaded, less RAM used etc. Also using different versions ends once passing types via method signatures happens (which is not the case here for ClassMate, though). So using ranges generally is preferrable IMO in order to reduce the number of different versions of a lib in use. As said it depends on the compatability of the libs of course. OSGi puts strong emphasize on the semantics of versions, so ranges should work from this perspective. It's a different story of course how lib authors stick to it in reality.